extensions/common/vault/vault-hashicorp/src/main/java/org/eclipse/edc/vault/hashicorp/HashicorpVaultClient.java

```diff
@@ -185,26 +288,85 @@ private HttpUrl getHealthUrl() {
 
-        final var vaultApiPath = hashicorpVaultConfig.getVaultApiSecretPath();
+        var vaultApiPath = configValues.secretPath();
 
-        return Objects.requireNonNull(HttpUrl.parse(hashicorpVaultConfig.getVaultUrl()))
+        return configValues.url()
                 .newBuilder()
                 .addPathSegments(PathUtil.trimLeadingOrEndingSlash(vaultApiPath))
                 .addPathSegment(entryType)
                 .addPathSegments(key)
                 .build();
     }
 
+    @NotNull
+    private Request httpGet(HttpUrl requestUri) {
+        return new Request.Builder()
+                .url(requestUri)
+                .headers(headers)
+                .get()
+                .build();
+    }
+
+    @NotNull
+    private Request httpPost(HttpUrl requestUri, Object requestBody) {
+        return new Request.Builder()
+                .url(requestUri)
+                .headers(headers)
+                .post(createRequestBody(requestBody))
+                .build();
+    }
+
+    @NotNull
+    private Headers getHeaders() {
+        var headersBuilder = new Headers.Builder().add(VAULT_REQUEST_HEADER, Boolean.toString(true));
+        headersBuilder.add(VAULT_TOKEN_HEADER, configValues.token());
+        return headersBuilder.build();
+    }
+
     private RequestBody createRequestBody(Object requestPayload) {
-        String jsonRepresentation = null;
+        String jsonRepresentation;
         try {
             jsonRepresentation = objectMapper.writeValueAsString(requestPayload);
         } catch (JsonProcessingException e) {
-            throw new RuntimeException(e);
+            throw new EdcException(e);
         }
         return RequestBody.create(jsonRepresentation, MEDIA_TYPE_APPLICATION_JSON);
     }
+
+    private Result<Boolean> parseRenewable(Map<String, Object> map) {
+        try {
+            var data = objectMapper.convertValue(getValueFromMap(map, DATA_KEY), new TypeReference<Map<String, Object>>() {
+            });
+            var isRenewable = objectMapper.convertValue(getValueFromMap(data, RENEWABLE_KEY), Boolean.class);
+            return Result.success(isRenewable);
+        } catch (IllegalArgumentException e) {
+            var errMsg = "Failed to parse renewable flag from %s with reason: %s".formatted(map, e.getMessage());
+            monitor.warning(errMsg);
+            return Result.failure(errMsg);
+        }
+    }
+
+    private Result<Long> parseTtl(Map<String, Object> map) {
+        try {
+            var auth = objectMapper.convertValue(getValueFromMap(map, AUTH_KEY), new TypeReference<Map<String, Object>>() {
+            });
+            var ttl = objectMapper.convertValue(getValueFromMap(auth, LEASE_DURATION_KEY), Long.class);
+            return Result.success(ttl);
+        } catch (IllegalArgumentException e) {
+            var errMsg = "Failed to parse ttl from %s with reason: %s".formatted(map, e.getMessage());
+            monitor.warning(errMsg);
```

Comment:
In this case, log the exception with a more descriptive error message (token renewal TTL) including the entire error so that the stack trace is preserved and return a simple failure. The caller should then simply abort the process. As it stands, the error is essentially logged twice, which does not add value to the output.